app/views/ChooseProvider.js

```diff
@@ -215,33 +199,59 @@ class ChooseProviderScreen extends Component {
     );
   }
 
+  toggleFilterAuthoritesByGPSHistory() {
+    this.filterAuthoritesByGPSHistory({
+      val: !this.state.isAuthorityFilterActive,
+    });
+  }
+
+  async filterAuthoritesByGPSHistory(isAuthorityFilterActive) {
+    await this.fetchAuthoritiesList(isAuthorityFilterActive.val);
+    this.setState({ isAuthorityFilterActive: isAuthorityFilterActive.val });
+  }
+
+  async toggleAutoSubscribe() {
+    this.setState(
+      prevState => ({
+        isAutoSubscribed: !prevState.isAutoSubscribed,
+      }),
+      async () => {
+        this.state.isAutoSubscribed
+          ? await HCAService.enableAutoSubscription()
+          : await HCAService.disableAutoSubscription();
+      },
+    );
+  }
+
   render() {
     return (
       <NavigationBarWrapper
         title={languages.t('label.choose_provider_title')}
         onBackPress={this.backToMain.bind(this)}>
         <View style={styles.main}>
-          <Typography style={styles.headerTitle}>
+          <Typography style={styles.headerTitle} use={Type.Headline2}>
             {languages.t('label.authorities_title')}
           </Typography>
-          <Typography style={styles.sectionDescription}>
+          <Typography style={styles.sectionDescription} use={Type.Body1}>
             {languages.t('label.authorities_desc')}
           </Typography>
+          <TouchableOpacity style={styles.autoSubcribe}>
+            <CheckBox
```

Comment:
Sam introduced a basic Checkbox in common/... we should converge. if we need custom UX/styles, let's use the one in common and make that better...

Would accept this for now to get this over the line. But would like to know the plan for reuse.

Reply:
Nice - didn't know we had that. I removed the package I pulled in and am using our custom Checkbox.